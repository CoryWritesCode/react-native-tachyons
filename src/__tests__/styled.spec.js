import { T, build, styled } from '../';
import * as React from 'react';
import TestRenderer from 'react-test-renderer';

build({});

const STYLE_CONST = 'jcc';

const StyledButton = styled('button')`flx-i flx-row ${STYLE_CONST} ${props => (props.primary ? 'blue' : 'white')}`;
const StyledButtonMultiLine = styled('button')`
  flx-i flx-row ${STYLE_CONST} ${props => (props.primary ? 'blue' : 'white')}
`;
const InlineButton = ({ primary, style }) => (
  <button primary={!!primary} style={T(`flx-i flx-row jcc ${primary ? 'blue' : 'white'}`, style)} />
);

test('styled (primary)', () => {
  const testStyle = { color: 'pink' };

  const testRenderer = TestRenderer.create(<StyledButton primary style={testStyle} />);

  expect(testRenderer.toJSON()).toMatchSnapshot();
});

test('styled multiLine', () => {
  const testStyle = { color: 'pink' };

  const testRenderer = TestRenderer.create(<StyledButtonMultiLine primary style={testStyle} />);

  expect(testRenderer.toJSON()).toMatchSnapshot();
});

test('styled (output is same as "normal inline library version")', () => {
  const testStyle = { color: 'pink' };

  const testRendererStyled = TestRenderer.create(<StyledButton primary style={testStyle} />);
  const testRendererInline = TestRenderer.create(<InlineButton primary style={testStyle} />);

  expect(testRendererInline.toJSON()).toEqual(testRendererStyled.toJSON());
});

test('with displayName', () => {
  const FunctionWithDisplayName = function() {
    return 'div';
  };
  FunctionWithDisplayName.displayName = 'FunctionWithDisplayNameDisplayName';
  const StyledFunctionWithDisplayName = styled(FunctionWithDisplayName)`pink`;

  expect(StyledFunctionWithDisplayName.displayName).toMatchSnapshot();
});

test('with name', () => {
  const FunctionWithName = function FunctionWithName() {
    return 'div';
  };
  const StyledFunctionWithName = styled(FunctionWithName)`pink`;

  expect(StyledFunctionWithName.displayName).toMatchSnapshot();
});

test('without name', () => {
  const StyledFunctionWithoutName = styled(function() {
    return 'div';
  })`pink`;

  expect(StyledFunctionWithoutName.displayName).toMatchSnapshot();
});

test('using withConfig()', () => {
  const FunctionWithName = function FunctionWithName() {
    return 'div';
  };
  const StyledFunctionWithName = styled(FunctionWithName).withConfig({
    displayName: 'doing this for our babel plugin'
  })`pink`;

  expect(StyledFunctionWithName.displayName).toMatchSnapshot();
});
