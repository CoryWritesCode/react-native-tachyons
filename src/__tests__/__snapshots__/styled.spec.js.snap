// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled (primary) 1`] = `
<button
  primary={true}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "color": "#0000ff",
      },
      Object {
        "color": "pink",
      },
    ]
  }
/>
`;

exports[`styled multiLine 1`] = `
<button
  primary={true}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "flexDirection": "row",
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "color": "#0000ff",
      },
      Object {
        "color": "pink",
      },
    ]
  }
/>
`;

exports[`using withConfig() 1`] = `"doing this for our babel plugin"`;

exports[`with displayName 1`] = `"StyledFunctionWithDisplayNameDisplayName"`;

exports[`with name 1`] = `"StyledFunctionWithName"`;

exports[`without name 1`] = `"StyledComponent"`;
